src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleControl.cs

```diff
@@ -2523,64 +2523,92 @@ internal static void SetConsoleWindowTitle(string consoleTitle)
+        /// </param>
+        /// <param name="newLine">
+        /// New line is written.
         /// </param>
         /// <exception cref="HostException">
-        /// if the Win32's WriteConsole fails
+        /// If the Win32's WriteConsole fails.
         /// </exception>
-
-        internal static void WriteConsole(ConsoleHandle consoleHandle, string output)
+        internal static void WriteConsole(ConsoleHandle consoleHandle, ReadOnlySpan<char> output, bool newLine)
         {
             Dbg.Assert(!consoleHandle.IsInvalid, "ConsoleHandle is not valid");
             Dbg.Assert(!consoleHandle.IsClosed, "ConsoleHandle is closed");
 
-            if (string.IsNullOrEmpty(output))
+            if (output.Length == 0)
+            {
+                if (newLine)
+                {
+                    WriteConsole(consoleHandle, ConsoleHostUserInterface.Crlf.AsSpan());
+                }
+
                 return;
+            }
 
             // Native WriteConsole doesn't support output buffer longer than 64K.
             // We need to chop the output string if it is too long.
-
             int cursor = 0; // This records the chopping position in output string
-            const int maxBufferSize = 16383; // this is 64K/4 - 1 to account for possible width of each character.
+            const int MaxBufferSize = 16383; // this is 64K/4 - 1 to account for possible width of each character.
 
             while (cursor < output.Length)
             {
-                string outBuffer;
+                ReadOnlySpan<char> outBuffer;
 
-                if (cursor + maxBufferSize < output.Length)
+                if (cursor + MaxBufferSize < output.Length)
                 {
-                    outBuffer = output.Substring(cursor, maxBufferSize);
-                    cursor += maxBufferSize;
+                    outBuffer = output.Slice(cursor, MaxBufferSize);
+                    cursor += MaxBufferSize;
                 }
                 else
                 {
-                    outBuffer = output.Substring(cursor);
+                    outBuffer = output.Slice(cursor);
                     cursor = output.Length;
                 }
 
-                DWORD charsWritten;
-                bool result =
-                    NativeMethods.WriteConsole(
-                        consoleHandle.DangerousGetHandle(),
-                        outBuffer,
-                        (DWORD)outBuffer.Length,
-                        out charsWritten,
-                        IntPtr.Zero);
-
-                if (result == false)
+                if (newLine)
                 {
-                    int err = Marshal.GetLastWin32Error();
-
-                    HostException e = CreateHostException(err, "WriteConsole",
-                        ErrorCategory.WriteError, ConsoleControlStrings.WriteConsoleExceptionTemplate);
-                    throw e;
+                    var endOfLine = ConsoleHostUserInterface.Crlf.AsSpan();
```

Comment:
Can we change `ConsoleHostUserInterface.Crlf` to `Environment.NewLine`?

Reply:
It is defined as `internal static string Crlf = System.Environment.NewLine;`.
If we want to replace we should replace everywhere (13 points). Make sense?

Reply:
`Crlf` is a bad name given we are x-plats now. I'm fine that we replace everywhere.

Reply:
Will do in follow PR.